components/List/tests/unit/MDCSelfSizingStereoCellTests.m

```diff
@@ -0,0 +1,157 @@
+  // Given
+  MDCTypographyScheme *typographyScheme = [[MDCTypographyScheme alloc] init];
+  MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
+  UIFont *titleFont = typographyScheme.subtitle1;
+  // TODO: Investigate what can be done about the line below failing when using .caption instead of
+  // .button. See
+  // https://github.com/material-components/material-components-ios/issues/7424 for
+  // context.
+  UIFont *detailFont = typographyScheme.button;
+  cell.titleLabel.font = titleFont;
+  cell.detailLabel.font = detailFont;
+  cell.mdc_adjustsFontForContentSizeCategory = YES;
+  cell.titleLabel.text = @"Title";
+  cell.detailLabel.text = @"Detail";
+  CGFloat defaultTitleSize = cell.titleLabel.font.pointSize;
+  CGFloat defaultDetailSize = cell.detailLabel.font.pointSize;
+
+  // When
+  MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+      *extraExtraLargeContainer =
+          [[MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow alloc]
+              initWithContentSizeCategoryOverride:UIContentSizeCategoryExtraLarge];
+  [extraExtraLargeContainer addSubview:cell];
+  [NSNotificationCenter.defaultCenter
+      postNotificationName:UIContentSizeCategoryDidChangeNotification
+                    object:nil];
+
+  // Then
+  // We only test the behavior for iOS10.0+ because we are not able to
+  // simulate UIApplication in this test for iOS9.0 path.
+  if (@available(iOS 10.0, *)) {
+    XCTAssertGreaterThan(cell.titleLabel.font.pointSize, defaultTitleSize);
+    XCTAssertGreaterThan(cell.detailLabel.font.pointSize, defaultDetailSize);
+  }
+}
+
+
+- (void)testAdjustsFontForContentSizeCategoryWhenScaledFontIsUnavailableIsYES {
+  // Given
+  MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
+  UIFont *originalTitleFont = [UIFont systemFontOfSize:99];
+  UIFont *originalDetailFont = [UIFont systemFontOfSize:99];
+  cell.titleLabel.font = originalTitleFont;
+  cell.detailLabel.font = originalDetailFont;
+  cell.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = YES;
+  cell.mdc_adjustsFontForContentSizeCategory = YES;
+  cell.titleLabel.text = @"Title";
+  cell.detailLabel.text = @"Detail";
+
+  // When
+  MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+  *extraExtraLargeContainer =
+  [[MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow alloc]
+   initWithContentSizeCategoryOverride:UIContentSizeCategoryExtraLarge];
+  [extraExtraLargeContainer addSubview:cell];
+  [NSNotificationCenter.defaultCenter
+   postNotificationName:UIContentSizeCategoryDidChangeNotification
+   object:nil];
+
+  // Then
+  XCTAssertFalse([cell.titleLabel.font mdc_isSimplyEqual:originalTitleFont]);
+  XCTAssertFalse([cell.detailLabel.font mdc_isSimplyEqual:originalDetailFont]);
+}
+
+- (void)testAdjustsFontForContentSizeCategoryWhenScaledFontIsUnavailableIsNO {
```

Comment:
testDoesNotAdjustFontForContentSizeCategoryWhenScaledFontIsUnavailableIsNO

Reply:
Addressed in d590fe6